sentinelhub/sentinelhub_catalog.py

```diff
@@ -188,6 +188,9 @@ def _get_data_collection_filters(data_collection):
         if data_collection.timeliness:
             filters['timeliness'] = {'eq': data_collection.timeliness}
 
+        if data_collection.dem_instance:
+            filters['demInstance'] = data_collection.dem_instance
```

Comment:
If I'm not mistaken Catalog API doesn't support DEM. If it ever will then the query would probably be something like: `{'eq': data_collection.dem_instance}`.

I suggest that for now this is removed. If support gets added we can add the correct filter then.